save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/TestSuitesSourceInternalController.kt

```diff
@@ -24,41 +27,62 @@ import reactor.kotlin.core.publisher.toMono
 class TestSuitesSourceInternalController(
     private val testSuitesSourceService: TestSuitesSourceService,
     private val testSuitesSourceVersionService: TestsSourceVersionService,
+    @Lazy
+    private val snapshotStorage: MigrationTestsSourceSnapshotStorage,
     private val organizationService: OrganizationService,
     private val executionService: ExecutionService,
     private val lnkExecutionTestSuiteService: LnkExecutionTestSuiteService,
 ) {
     /**
-     * @param versionInfo
+     * @param snapshotInfo
      * @param contentAsMonoPart
      * @return [Mono] without value
      */
     @PostMapping("/upload-snapshot", consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
     fun uploadSnapshot(
-        @RequestPart("versionInfo") versionInfo: TestsSourceVersionInfo,
+        @RequestPart("snapshotInfo") snapshotInfo: TestsSourceSnapshotInfo,
         @RequestPart("content") contentAsMonoPart: Mono<Part>,
     ): Mono<Unit> = contentAsMonoPart.flatMap { part ->
+        val key = TestSuitesSourceSnapshotKey(
+            organizationName = snapshotInfo.organizationName,
+            testSuitesSourceName = snapshotInfo.sourceName,
+            version = snapshotInfo.commitId,
+            creationTime = snapshotInfo.commitTime,
+        )
         val content = part.content().map { it.asByteBuffer() }
-        testSuitesSourceVersionService.upload(versionInfo, content).map { writtenBytes ->
-            log.info { "Saved ($writtenBytes bytes) snapshot of ${versionInfo.sourceName} in ${versionInfo.organizationName} with version ${versionInfo.version}" }
+        snapshotStorage.upload(key, content).map { writtenBytes ->
```

Comment:
Could we move such logging into the storage class directly? I think we will need it for any type of `upload/copy/etc`

Reply:
moved